### PR TITLE
HvH mode, civil war

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

This is a first batch of the HvH mode. It is playable-ish, it still lacks things like objectives or even a balance function.

WARNING: this is only playable on the ship map = twin pillars and on the ground map ice colony

This adds to the other PR:

The twin pillars shipmap, it's a map with two identical ships, one for the rebel faction, the other for the loyalist faction
Some small refactoring of the join system, so you are able to join the loyalist or rebel faction. This is very much WIP and will be improved in later PR, but this is good enough for some test
Another shuttle, the triumph. It's a copy of the alamo, but for the rebels.
Add the energy weapon in HvH vendors since david forgot to do it

This is meant for one or two tests rounds with probably a lot of admin intervention, as a lot is to be tested.

## Why It's Good For The Game

New gamemode

## Changelog
:cl:
add: New HvH gamemode, civil war. Very much WIP
/:cl:

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
This looks interesting. Can't wait to see how it functions in-game. Hopefully it gets to a state where it can be a common occurrence

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -339,6 +339,9 @@ SUBSYSTEM_DEF(job)
 	if(assigned_role && length(GLOB.jobspawn_overrides[assigned_role])) //We're doing something special today.
 		SendToAtom(M, pick(GLOB.jobspawn_overrides[assigned_role]))
 		return
+	if(assigned_role.faction == FACTION_TERRAGOV_REBEL && length(GLOB.latejoinrebel))
+		SendToAtom(M, pick(GLOB.latejoinrebel))
+		return
```
Comment (SplinterGP) [inline code/controllers/subsystem/job.dm:344]: cant this be done other way, giving latejoin markers factions or something ?

File: code/game/objects/effects/landmarks/landmarks.dm [hunk 2]
```diff
@@ -387,3 +392,25 @@
...
+
+//used to spawn a different dropship control console groundside for HvH mode
+/obj/effect/landmark/dropship_console_spawn_lz1
+	name = "Dropship console spawn lz1"
+	icon = 'icons/obj/machines/computer.dmi'
+	icon_state = "shuttle"
+
+/obj/effect/landmark/dropship_console_spawn_lz1/Initialize()
+	GLOB.lz1_shuttle_console_turfs_list += loc
+	..()
```
Comment (SplinterGP) [inline code/game/objects/effects/landmarks/landmarks.dm:404]: why here and not before ?

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/datums/gamemodes/game_mode.dm [hunk 3]
```diff
@@ -57,6 +59,17 @@
...
+	if(flags_landmarks & MODE_LANDMARK_SPAWN_SPECIFIC_SHUTTLE_CONSOLE)
+		for(var/turf/T AS in GLOB.lz1_shuttle_console_turfs_list)
+			new /obj/machinery/computer/shuttle/shuttle_control/dropship/rebel(T)
+		for(var/turf/T AS in GLOB.lz2_shuttle_console_turfs_list)
+			new /obj/machinery/computer/shuttle/shuttle_control/dropship/loyalist(T)
+	else
+		for(var/turf/T AS in GLOB.lz1_shuttle_console_turfs_list)
+			new /obj/machinery/computer/shuttle/shuttle_control/dropship(T)
+		for(var/turf/T AS in GLOB.lz2_shuttle_console_turfs_list)
+			new /obj/machinery/computer/shuttle/shuttle_control/dropship(T)
```
Comment (SplinterGP) [inline code/datums/gamemodes/game_mode.dm:71]:
```suggestion
		for(var/turf/T AS in GLOB.lz1_shuttle_console_turfs_list + GLOB.lz2_shuttle_console_turfs_list)
			new /obj/machinery/computer/shuttle/shuttle_control/dropship(T)
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**